### PR TITLE
JavaScript-II Project

### PR DESCRIPTION
**Pull request by Kellswork**

### CODE REVIEW

File: assignments/array-methods.js [hunk 1]
```diff
@@ -56,28 +56,55 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
...
+ runners.forEach(runner => {
+     fullName.push(`${runner.first_name} ${runner.last_name}`);
+ });
+
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
+allcaps = runners.map(runner => {
```
Comment (GianBarGian) [inline assignments/array-methods.js:68]: Be careful, you have a typo here. Try to check that your code is running before committing to GitHub.

File: assignments/array-methods.js [hunk 2]
```diff
@@ -56,28 +56,55 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
...
+});
+
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
 let largeShirts = [];
+largeShirts = runners.filter(runner => {
+   return runner.shirt_size.toLowerCase().indexOf(('L').toLowerCase()) > -1;
+});
```
Comment (GianBarGian) [inline assignments/array-methods.js:79]: This is not returning just "L" size runners, but even XL 2XL etc..

File: assignments/callbacks.js [hunk 3]
```diff
@@ -1,55 +1,85 @@
...
 */
 
+// function getLength(arr, cb) {
+//   // getLength passes the length of the array into the callback.
+//   return cb(arr.length);
+// }
+// // function invocation
+// getLength(items, function(length) {
+//  console.log("Length of array:", length);
+// });
```
Comment (GianBarGian) [inline assignments/callbacks.js:32]: Good job with callbacks!

File: assignments/closure.js [hunk 4]
```diff
@@ -1,21 +1,43 @@
...
+  }
+  return sayHi();
+}
+ console.log(greeting());
 
 /* STRETCH PROBLEMS, Do not attempt until you have completed all previous tasks for today's project files */
 
-
 // ==== Challenge 2: Create a counter function ====
 const counter = () => {
```
Comment (GianBarGian) [inline assignments/closure.js:16]: Good job with stretch goals on closures!

**Comment by GianBarGian** [comment]
Good job with the project. You have a couple of small mistakes in your array-methods file but apart from those your project looks really good.
Well done with callbacks and closures!

File: assignments/array-methods.js (hunk 2 above)
Comment (Kellswork) [reply assignments/array-methods.js:79]: Oh Okay I thought It was supposed to return all large sized shirts including XL 2XL. My mistake

**Pull request merged**